core/services/cable_guy/api/manager.py

```diff
@@ -396,71 +415,182 @@ def get_interfaces_priority(self) -> List[NetworkInterfaceMetric]:
+            ],
+            key=lambda x: x.priority,
+            reverse=True,
+        )
+
         return [
-            NetworkInterfaceMetric(index=index, name=name, priority=metric_dict.get(index) or 0)
-            for index, name in name_dict.items()
+            NetworkInterfaceMetric(index=item.index, name=item.name, priority=original_list[-(i + 1)].priority)
+            for i, item in enumerate(original_list)
         ]
 
-    def get_interface_priority(self, interface_name: str) -> Optional[NetworkInterfaceMetric]:
-        """Get the priority metric for a network interface.
+    def _get_service_dhcpcd_content(self) -> List[str]:
+        """Returns a list of lines from the dhcpcd configuration file that belong to
+        this service.
+        Any exceptions are caught and logged, and an empty list is returned.
 
-        Args:
-            interface_name (str): The name of the network interface.
+        List[str]: Lines that will be used by this service
+        """
+        try:
+            with open(self.dhcpcd_conf_path, "r", encoding="utf-8") as f:
+                lines = f.readlines()
+
+                start, end = None, None
+                for i, line in enumerate(lines):
+                    # Get always the first occurrence of 'start' and last of 'end'
+                    if self.dhcpcd_conf_start_string in line and start is None:
+                        start = i
+                    if self.dhcpcd_conf_end_string in line:
+                        end = i
+
+                # Remove everything that is not from us
+                if start is not None and end is not None:
+                    del lines[0 : start + 1]
+                    del lines[end:-1]
+
+                # Clean all lines and remove empty ones
+                lines = [line.strip() for line in lines]
+                lines = [line for line in lines if line]
+                return lines
+        except Exception as exception:
+            logger.warning(f"Failed to read {self.dhcpcd_conf_path}, error: {exception}")
+            return []
 
-        Returns:
-            Optional[NetworkInterfaceMetric]: The priority metric for the interface, or None if no metric found.
+    def _get_interface_priority_from_dhcpcd(self) -> List[NetworkInterfaceMetric]:
+        """Parses dhcpcd config file to get network interface priorities.
+        Goes through the dhcpcd config file line by line looking for "interface"
+        and "metric" lines. Extracts the interface name and metric value. The
+        metric is used as the priority, with lower being better.
+
+        List[NetworkInterfaceMetric]: A list of priority metrics for each interface.
         """
-        metric: NetworkInterfaceMetric
-        for metric in self.get_interfaces_priority():
-            if interface_name == metric.name:
-                return metric
+        lines = self._get_service_dhcpcd_content()
+        result = []
+        current_interface = None
```

Comment:
did you miss the type here? it looks like you missed the type, and that caused the errors in line 480 and 482 that you suppressed.

Reply:
Yes, this is unrelated, the source of the issue is from mypy.
https://github.com/python/mypy/issues/14987